compiler/rustc_index_macros/Cargo.toml

```diff
@@ -7,7 +7,7 @@ edition = "2021"
 proc-macro = true
 
 [dependencies]
-syn = { version = "2.0.9", features = ["full"] }
+syn = { version = "2.0.9", features = ["full", "extra-traits"] }
```

Comment:
Why do you need to enable the feature here and in `compiler/rustc_type_ir_macros/Cargo.toml`? Did it work with resolver 1 but not resolver 2?

Reply:
Due to how resolver 2's feature unification works, resolver 1 always unifies across the workspace, resolver 2 only across the packages being built (which is the entire problem we were having as upstream has set up the features incorrectly).

This change will be upstreamed afterwards.